documentation/create_overview_doc/createdoc.sh

```diff
@@ -570,46 +340,25 @@ function main() {
         fi
     done
 
-    # Extract content and call Bito for a system introduction and summary
-    local introduction_and_summary=$(extract_module_names_and_associated_objectives_then_call_bito "$aggregated_md_file" "$prompt_folder/system_introduction_prompt.txt")
+    local introduction_and_summary=$(extract_module_details_and_call_bito "$aggregated_md_file" "$prompt_folder/system_introduction_prompt.txt")
 
-    # Prepend the introduction and summary to the aggregated markdown file
-    # Save the current content of the aggregated file temporarily
+    # Prepend the introduction and summary to the aggregated Markdown file
     local temp_file=$(mktemp)
     mv "$aggregated_md_file" "$temp_file"
-
-    # Create a new aggregated file starting with the Markdown-formatted introduction title
     echo -e "# Introduction :\n" > "$aggregated_md_file"
-
-    # Append the introduction and summary
     echo "$introduction_and_summary" >> "$aggregated_md_file"
 
-    # Call generate_mdd_overview function here, after all .mdd files are created
-    generate_mdd_overview "$docs_folder" "$aggregated_md_file"
-    
-    # Append the content of overview.mdd to the aggregated file extracting only the Mermaid diagram block
-    if [ -f "$docs_folder/overview.mdd" ]; then
-        echo -e "\n# Full System Overview\n" >> "$aggregated_md_file"
-
-        # Extract and append only the Mermaid diagram block
-        awk '/^```mermaid$/,/^```$/' "$docs_folder/overview.mdd" >> "$aggregated_md_file"
-    else
-        echo "Overview file not found or empty."
-    fi
+    # Generate and append the system flow chart
+    generate_flow_chart_for_system "$folder_to_document" "$docs_folder/flowmaps/system_flowchart.png"
+    echo -e "\n# System Flowchart\n\n![System Flowchart](./flowmaps/system_flowchart.png)\n" >> "$aggregated_md_file"
 
     # Append the rest of the original aggregated content
     cat "$temp_file" >> "$aggregated_md_file"
-
-    # Remove the temporary file
     rm "$temp_file"
```

Comment:
**Performance Issue**: The function 'extract_module_details_and_call_bito' is expected to extract module details and call Bito for generating a system introduction and summary. However, the implementation details of this function are not provided in the diff, which could potentially lead to performance issues if not properly implemented. Specifically, if the function makes inefficient use of Bito calls or processes large amounts of data inefficiently, it could significantly slow down the documentation generation process. <br> **Fix**: Ensure that 'extract_module_details_and_call_bito' function is optimized for performance. This includes minimizing the number of Bito calls required, efficiently processing data within the function, and avoiding unnecessary computation. Consider caching results of Bito calls if they can be reused, and thoroughly test the function with large datasets to identify any potential bottlenecks. <br> **Code Suggestion**: 
 ```
 extract_module_details_and_call_bito() {
   # Optimize data processing
   local module_details=$(echo "$1" | jq '.[] | select(.module == "target_module")')
   # Minimize Bito calls
   if [[ -z $(echo "$module_details" | grep 'processed') ]]; then
     call_bito "$module_details"
   fi
 }
 ```